### DISCUSSION TITLE
Depoloyment 3.3.1, still not seccessful

### DISCUSSION THREAD

**Discussion by monbasa** [cobbler/cobbler#2933] [Version specific questions] [answered]
### 1.OS info

	[root@cobb3 ~]# lsb_release -a
	LSB Version:    :core-4.1-amd64:core-4.1-noarch
	Distributor ID: Fedora
	Description:    Fedora release 35 (Thirty Five)
	Release:        35
	Codename:       ThirtyFive

	[root@cobb3 ~]# uname -a
	Linux cobb3 5.14.10-300.fc35.x86_64 #1 SMP Thu Oct 7 20:48:44 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### 2.Cobbler Verison

	# cobbler version
	Cobbler 3.3.1
	  source: e36be05e, Fri Feb 4 15:58:10 2022 +0100
	  build time: Thu Feb 10 06:27:59 2022
  
#### 3.1.Perpare software env

maybe forget some package to write as below

	dnf install -y syslinux python3-pip rpm-build httpd dhcp tftp tftp-server

	dnf install -y python3-cheetah python3-coverage python3-devel python3-netaddr python3-schema python3-sphinx python3-mod_wsgi python3-dns python3-file-magic
	pip3 install sphinx coverage pyflakes pycodestyle wheel wsgi_module 

	dnf install yum-utils pykickstart fence-agents createrepo debmirror

#### 3.2.make cobbler 

	#make webtest

#### 3.3.Start HTTP / TFTP service

	systemctl enable httpd tftp --now

#### 3.4.modify "cobblerd.service" as below

	/usr/bin/cobblerd  >>> /usr/local/cobblerd.service

	cp /etc/cobbler/cobblerd.service /etc/systemd/system/cobbberd

#### 3.5.Start cobbler service

	systemctl enable cobblerd --now

#### 3.6.modify /etc/cobbler/setting.yaml, I only modify as below

		allow_dynamic_settings: true

		default_password_crypted: $1$random-p$a.dCG5liiKkewgkxqL34v/

		next_server_v4: 192.168.80.1

		server: 192.168.80.1

		windows_enabled: true

		manage_dhcp: true

		manage_dhcp_v4: true

#### 3.7.config /etc/cobbler/dhcp.templete,modify as below

		subnet 192.168.80.0 netmask 255.255.255.0 {
			 option routers             192.168.80.1;
			 option domain-name-servers 192.168.80.1;
			 option subnet-mask         255.255.255.0;
			 range dynamic-bootp        192.168.80.5 192.168.80.10;
			 default-lease-time         21600;
			 max-lease-time             43200;
			 next-server                $next_server_v4;
		 
#### 3.8.run "cobbler check", as below result

		# cobbler check
		The following are potential configuration items that you may want to fix:

		1: For PXE to be functional, the 'next_server_v6' field in /etc/cobbler/settings.yaml must be set to something other than ::1, and should match the IP of the boot server on the PXE network.
		2: some network boot-loaders are missing from /var/lib/cobbler/loaders. If you only want to handle x86/x86_64 netbooting, you may ensure that you have installed a *recent* version of the syslinux package installed and can ignore this message entirely. Files in this directory, should you want to support all architectures, should include pxelinux.0, andmenu.c32.
		3: comment out 'dists' on /etc/debmirror.conf for proper debian support
		4: comment out 'arches' on /etc/debmirror.conf for proper debian support

		Restart cobblerd and then run 'cobbler sync' to apply changes.

#### 3.10.check net port

		# ss -tunlp
		Netid         State          Recv-Q          Send-Q                   Local Address:Port                    Peer Address:Port         Process
		udp           UNCONN         0               0                              0.0.0.0:5355                         0.0.0.0:*             users:(("systemd-resolve",pid=847,fd=10))
		udp           UNCONN         0               0                              0.0.0.0:50426                        0.0.0.0:*             users:(("dhcpd",pid=1585236,fd=20))
		udp           UNCONN         0               0                        127.0.0.53%lo:53                           0.0.0.0:*             users:(("systemd-resolve",pid=847,fd=16))
		udp           UNCONN         0               0                              0.0.0.0:67                           0.0.0.0:*             users:(("dhcpd",pid=1585236,fd=7))
		udp           UNCONN         0               0                            127.0.0.1:323                          0.0.0.0:*             users:(("chronyd",pid=1512717,fd=5))
		udp           UNCONN         0               0                                 [::]:5355                            [::]:*             users:(("systemd-resolve",pid=847,fd=12))
		udp           UNCONN         0               0                                 [::]:42702                           [::]:*             users:(("dhcpd",pid=1585236,fd=21))
		udp           UNCONN         0               0                                    *:69                                 *:*             users:(("systemd",pid=1,fd=96))
		udp           UNCONN         0               0                                [::1]:323                             [::]:*             users:(("chronyd",pid=1512717,fd=6))
		tcp           LISTEN         0               4096                           0.0.0.0:5355                         0.0.0.0:*             users:(("systemd-resolve",pid=847,fd=11))
		tcp           LISTEN         0               64                           127.0.0.1:45295                        0.0.0.0:*             users:(("cockpit-bridge",pid=1512362,fd=9))
		tcp           LISTEN         0               4096                     127.0.0.53%lo:53                           0.0.0.0:*             users:(("systemd-resolve",pid=847,fd=17))
		tcp           LISTEN         0               128                            0.0.0.0:22                           0.0.0.0:*             users:(("sshd",pid=951,fd=3))
		tcp           LISTEN         0               5                            127.0.0.1:25151                        0.0.0.0:*             users:(("cobblerd",pid=1488957,fd=4))
		tcp           LISTEN         0               4096                              [::]:5355                            [::]:*             users:(("systemd-resolve",pid=847,fd=13))
		tcp           LISTEN         0               511                                  *:80                                 *:*             users:(("httpd",pid=1583518,fd=4),("httpd",pid=1583517,fd=4),("httpd",pid=1583516,fd=4),("httpd",pid=1583514,fd=4))
		tcp           LISTEN         0               128                               [::]:22                              [::]:*             users:(("sshd",pid=951,fd=4))
		tcp           LISTEN         0               511                                  *:443                                *:*             users:(("httpd",pid=1583518,fd=6),("httpd",pid=1583517,fd=6),("httpd",pid=1583516,fd=6),("httpd",pid=1583514,fd=6))
		tcp           LISTEN         0               4096                                 *:9090                               *:*             users:(("cockpit-tls",pid=1512262,fd=3),("systemd",pid=1,fd=79))

#### 3.11. Try HP DL360 G7 server boot from PXE

we can got IP, and tftp not found file

#### 3.12. run "cobbler mkloaders" and reboot client

Grub menu only prompt item "lcoal"

#### 3.13. Import OS

cobbler import --path=/mnt/ --name=CentOS-Linux-7.9-x86_64-Mini --arch=x86_64
cobbler import --path=/mnt --name=Fedora-Server-dvd-35-1.2-x86_64 --arch=x86_64

#### 3.15. modify /etc/cobbler/dhcp.templete as below 

		 class "pxeclients" {
				  match if substring (option vendor-class-identifier, 0, 9) = "PXEClient";

				  # Legacy
				  if option system-arch = 00:00 {
					  filename "pxelinux.0";				#<<<<<<<  modify 
				  }
				  # UEFI-32-1
				  if option system-arch = 00:06 {
					   # Not supported, no 32 bit UEFI grub executable
					  filename "unsupported";
				  }
				  # UEFI-32-2
				  if option system-arch = 00:02 {
					  # Not supported, no 32 bit UEFI grub executable
					  filename "unsupported";
				  }
				  # UEFI-64-1
				  else if option system-arch = 00:07 {
					  filename "grub/grubx64.efi";
				  }
				  # UEFI-64-2
				  else if option system-arch = 00:08 {
					  filename "grub/grubx64.efi";
				  }
				  # UEFI-64-3
				  else if option system-arch = 00:09 {
					  filename "grub/grubx64.efi";
				  }
				  # armv7   (aka arm 32 bit)
				  else if option system-arch = 00:0a {
					  filename "grub/armv7.efi";
				  }
				  # aarch64 (aka arm 64 bit)
				  else if option system-arch = 00:0b {
					  filename "grub/grubaa64.efi";
				  }
				  # RiskV 32 bit
				  else if option system-arch = 00:25 {
					  #ToDo petitboot loader
					  # petitboot should support pxelinux config files
					  filename "unsupported";
				  }
				  #RiskV 32 bit
				  else if option system-arch = 00:27 {
					  #ToDo petitboot loader
					  # petitboot should support pxelinux config files
					  filename "unsupported";
				  }
				  else if option system-arch = 00:0e {
					  # FIXME add petitboot support for ppc64(le)
					  filename "grub/grub.ppc64le";
				  }
				  else
				  {
					  filename "pxelinux.0"; 			#<<<<<<<  modify 
				  }
			 }
		}

#### 3.16.reboot client again, grub menu have all new item, but can not finish install

So, I do not kown, the cobbler 3.3.1 is a complete version. Or I make a mistake?

**Comment by monbasa** [comment]
updates firewall rule

	# firewall-cmd --zone=cobb --list-all
	cobb (active)
	  target: default
	  icmp-block-inversion: no
	  interfaces: em4
	  sources:
	  services: dhcp http https tftp
	  ports:
	  protocols:
	  forward: no
	  masquerade: no
	  forward-ports:
	  source-ports:
	  icmp-blocks:
	  rich rules:

**Comment by SchoolGuy** [comment]
So we are currently developing Cobbler 3.3.1 thus this is from the `master` branch.  Thus bugs are expected. We are not yet there to have a always stable rolling release.

Still what you are trying to do should always work on `master`. Thus a few notes and questions:

- If you import a distro you need to leave it mounted. If you unmount it Cobbler won't be able to function correctly.
- Since you got an IP correctly I would expect the DHCP config to be correct since then you have a specific entry for said system.
- I would check first if the file you are requesting is present in the TFTP Root at `/var/lib/tftproot`. (Exact location may differ.)

**Comment by monbasa** [comment]
```
[root@cobb3 cobbler]# cd /var/lib/tftpboot/
[root@cobb3 tftpboot]# cobbler distro report
Name                           : CentOS-Linux-7.9-Mini-x86_64
Architecture                   : x86_64
Automatic Installation Template Metadata : {}
TFTP Boot Files                : {}
Boot loaders                   : <<inherit>>
Breed                          : redhat
Comment                        :
Fetchable Files                : {}
Initrd                         : /var/www/cobbler/distro_mirror/CentOS-Linux-7.9-x86_64-Mini-x86_64/images/pxeboot/initrd.img
Kernel                         : /var/www/cobbler/distro_mirror/CentOS-Linux-7.9-x86_64-Mini-x86_64/images/pxeboot/vmlinuz
Kernel Options                 : {}
Kernel Options (Post Install)  : {}
Management Classes             : []
OS Version                     : rhel7
Owners                         : ['admin']
Redhat Management Key          :
Remote Boot Initrd             :
Remote Boot Kernel             :
Template Files                 : {}

[root@cobb3 tftpboot]# cobbler profile report
Name                           : CentOS-Linux-7.9-Mini-x86_64
Automatic Installation Template : sample.ks
Automatic Installation Metadata : <<inherit>>
TFTP Boot Files                : <<inherit>>
Boot loaders                   : <<inherit>>
Comment                        :
DHCP Tag                       :
Distribution                   : CentOS-Linux-7.9-Mini-x86_64
Enable iPXE?                   : False
Enable PXE Menu?               : True
Fetchable Files                : <<inherit>>
DHCP Filename Override         :
Kernel Options                 : <<inherit>>
Kernel Options (Post Install)  : <<inherit>>
Parent boot menu               :
Management Classes             : <<inherit>>
Management Parameters          : <<inherit>>
Name Servers                   : []
Name Servers Search Path       : []
Next Server (IPv4) Override    : 192.168.80.1
Next Server (IPv6) Override    : ::1
Owners                         : ['admin']
Parent Profile                 :
Proxy                          : <<inherit>>
Red Hat Management Key         : <<inherit>>
Repos                          : []
Server Override                : <<inherit>>
Template Files                 : {}
Virt Auto Boot                 : True
Virt Bridge                    : xenbr0
Virt CPUs                      : 1
Virt Disk Driver Type          : raw
Virt File Size(GB)             : 0
Virt Path                      :
Virt RAM (MB)                  : 512
Virt Type                      : kvm

[root@cobb3 tftpboot]# pwd
/var/lib/tftpboot
[root@cobb3 tftpboot]# ls -R
.:
grub  grub.cfg  images  ipxe  ldlinux.c32  libutil.c32  memdisk  menu.c32  pxelinux.0  pxelinux.cfg

./grub:
grub.0    i386-pc  local_efi.cfg     local_powerpc-ieee1275.cfg  system
grub.cfg  images   local_legacy.cfg  shim.efi                    system_link

./grub/i386-pc:
acpi.mod                     font.mod                  macbless.mod         relocator.mod
adler32.mod                  freedos.mod               macho.mod            romfs.mod
affs.mod                     fshelp.mod                mda_text.mod         scsi.mod
afs.mod                      fs.lst                    mdraid09_be.mod      search_fs_file.mod
afsplitter.mod               functional_test.mod       mdraid09.mod         search_fs_uuid.mod
ahci.mod                     gcry_arcfour.mod          mdraid1x.mod         search_label.mod
all_video.mod                gcry_blowfish.mod         memdisk.mod          search.mod
aout.mod                     gcry_camellia.mod         memrw.mod            sendkey.mod
appendedsig.mod              gcry_cast5.mod            minicmd.mod          serial.mod
appended_signature_test.mod  gcry_crc.mod              minix2_be.mod        setjmp.mod
archelp.mod                  gcry_des.mod              minix2.mod           setjmp_test.mod
asn1.mod                     gcry_dsa.mod              minix3_be.mod        setpci.mod
ata.mod                      gcry_idea.mod             minix3.mod           sfs.mod
at_keyboard.mod              gcry_md4.mod              minix_be.mod         shift_test.mod
backtrace.mod                gcry_md5.mod              minix.mod            signature_test.mod
bfs.mod                      gcry_rfc2268.mod          mmap.mod             sleep.mod
biosdisk.mod                 gcry_rijndael.mod         moddep.lst           sleep_test.mod
bitmap.mod                   gcry_rmd160.mod           modinfo.sh           smbios.mod
bitmap_scale.mod             gcry_rsa.mod              morse.mod            spkmodem.mod
blocklist.mod                gcry_seed.mod             mpi.mod              squash4.mod
blscfg.mod                   gcry_serpent.mod          msdospart.mod        strtoull_test.mod
boot_hybrid.img              gcry_sha1.mod             mul_test.mod         syslinuxcfg.mod
boot.img                     gcry_sha256.mod           multiboot2.mod       tar.mod
boot.mod                     gcry_sha512.mod           multiboot.mod        terminal.lst
bsd.mod                      gcry_tiger.mod            nativedisk.mod       terminal.mod
bswap_test.mod               gcry_twofish.mod          net.mod              terminfo.mod
btrfs.mod                    gcry_whirlpool.mod        newc.mod             test_asn1.mod
bufio.mod                    gdb_grub                  nilfs2.mod           test_blockarg.mod
cat.mod                      gdb.mod                   normal.mod           testload.mod
cbfs.mod                     geli.mod                  ntfscomp.mod         test.mod
cbls.mod                     gettext.mod               ntfs.mod             testspeed.mod
cbmemc.mod                   gfxmenu.mod               ntldr.mod            tftp.mod
cbtable.mod                  gfxterm_background.mod    odc.mod              tga.mod
cbtime.mod                   gfxterm_menu.mod          offsetio.mod         time.mod
cdboot.img                   gfxterm.mod               ohci.mod             trig.mod
chain.mod                    gmodule.pl                part_acorn.mod       tr.mod
cmdline_cat_test.mod         gptsync.mod               part_amiga.mod       truecrypt.mod
cmosdump.mod                 gzio.mod                  part_apple.mod       true.mod
cmostest.mod                 halt.mod                  part_bsd.mod         udf.mod
cmp.mod                      hashsum.mod               part_dfly.mod        ufs1_be.mod
cmp_test.mod                 hdparm.mod                part_dvh.mod         ufs1.mod
command.lst                  hello.mod                 part_gpt.mod         ufs2.mod
configfile.mod               help.mod                  partmap.lst          uhci.mod
config.h                     hexdump.mod               part_msdos.mod       usb_keyboard.mod
cpio_be.mod                  hfs.mod                   part_plan.mod        usb.mod
cpio.mod                     hfspluscomp.mod           part_sun.mod         usbms.mod
cpuid.mod                    hfsplus.mod               part_sunpc.mod       usbserial_common.mod
crc64.mod                    http.mod                  parttool.lst         usbserial_ftdi.mod
cryptodisk.mod               increment.mod             parttool.mod         usbserial_pl2303.mod
crypto.lst                   iorw.mod                  password.mod         usbserial_usbdebug.mod
crypto.mod                   iso9660.mod               password_pbkdf2.mod  usbtest.mod
cs5536.mod                   jfs.mod                   pata.mod             vbe.mod
ctz_test.mod                 jpeg.mod                  pbkdf2.mod           version.mod
datehook.mod                 json.mod                  pbkdf2_test.mod      vga.mod
date.mod                     kernel.exec               pcidump.mod          vga_text.mod
datetime.mod                 kernel.img                pci.mod              video_bochs.mod
diskboot.img                 keylayouts.mod            pgp.mod              video_cirrus.mod
diskfilter.mod               keystatus.mod             pkcs1_v15.mod        video_colors.mod
disk.mod                     ldm.mod                   plan9.mod            video_fb.mod
div.mod                      legacycfg.mod             play.mod             videoinfo.mod
div_test.mod                 legacy_password_test.mod  png.mod              video.lst
dm_nv.mod                    linux.mod                 priority_queue.mod   video.mod
drivemap.mod                 lnxboot.image             probe.mod            videotest_checksum.mod
echo.mod                     lnxboot.img               procfs.mod           videotest.mod
efiemu.mod                   loadenv.mod               progress.mod         wrmsr.mod
ehci.mod                     loopback.mod              pxeboot.img          xfs.mod
elf.mod                      lsacpi.mod                pxechain.mod         xnu.mod
eval.mod                     lsapm.mod                 pxe.mod              xnu_uuid.mod
exfat.mod                    lsmmap.mod                raid5rec.mod         xnu_uuid_test.mod
exfctest.mod                 ls.mod                    raid6rec.mod         xzio.mod
ext2.mod                     lspci.mod                 random.mod           zfscrypt.mod
extcmd.mod                   luks2.mod                 rdmsr.mod            zfsinfo.mod
f2fs.mod                     luks.mod                  read.mod             zfs.mod
fat.mod                      lvm.mod                   reboot.mod           zstd.mod
fdt.lst                      lzma_decompress.img       regexp.mod
file.mod                     lzopio.mod                reiserfs.mod

./grub/system:

./grub/system_link:

./images:
CentOS-Linux-7.9-Mini-x86_64

./images/CentOS-Linux-7.9-Mini-x86_64:
initrd.img  vmlinuz

./ipxe:
default.ipxe

./pxelinux.cfg:
default
[root@cobb3 tftpboot]#
```

**Comment by monbasa** [comment]
```
[root@cobb3 tftpboot]# cat /etc/dhcp/dhcpd.conf
# ******************************************************************
# Cobbler managed dhcpd.conf file
# generated from cobbler dhcp.conf template (Thu Feb 10 08:45:05 2022)
# Do NOT make changes to /etc/dhcpd.conf. Instead, make your changes
# in /etc/cobbler/dhcp.template, as /etc/dhcpd.conf will be
# overwritten.
# ******************************************************************

ddns-update-style interim;

allow booting;
allow bootp;

ignore client-updates;
set vendorclass = option vendor-class-identifier;

option system-arch code 93 = unsigned integer 16;

subnet 192.168.80.0 netmask 255.255.255.0 {
     option routers             192.168.80.1;
     option domain-name-servers 192.168.80.1;
     option subnet-mask         255.255.255.0;
     range dynamic-bootp        192.168.80.5 192.168.80.10;
     default-lease-time         21600;
     max-lease-time             43200;
     next-server                192.168.80.1;
     class "pxeclients" {
          match if substring (option vendor-class-identifier, 0, 9) = "PXEClient";

          # Legacy
          if option system-arch = 00:00 {
              filename "pxelinux.0";
          }
          # UEFI-32-1
          if option system-arch = 00:06 {
               # Not supported, no 32 bit UEFI grub executable
              filename "unsupported";
          }
          # UEFI-32-2
          if option system-arch = 00:02 {
              # Not supported, no 32 bit UEFI grub executable
              filename "unsupported";
          }
          # UEFI-64-1
          else if option system-arch = 00:07 {
              filename "grub/grubx64.efi";
          }
          # UEFI-64-2
          else if option system-arch = 00:08 {
              filename "grub/grubx64.efi";
          }
          # UEFI-64-3
          else if option system-arch = 00:09 {
              filename "grub/grubx64.efi";
          }
          # armv7   (aka arm 32 bit)
          else if option system-arch = 00:0a {
              filename "grub/armv7.efi";
          }
          # aarch64 (aka arm 64 bit)
          else if option system-arch = 00:0b {
              filename "grub/grubaa64.efi";
          }
          # RiskV 32 bit
          else if option system-arch = 00:25 {
              #ToDo petitboot loader
              # petitboot should support pxelinux config files
              filename "unsupported";
          }
          #RiskV 32 bit
          else if option system-arch = 00:27 {
              #ToDo petitboot loader
              # petitboot should support pxelinux config files
              filename "unsupported";
          }
          else if option system-arch = 00:0e {
              # FIXME add petitboot support for ppc64(le)
              filename "grub/grub.ppc64le";
          }
          else
          {
              filename "pxelinux.0";
          }
     }
}

    # group for Cobbler DHCP tag: default
    group {
}
```

**Comment by monbasa** [comment]
```
# systemctl status tftp
○ tftp.service - Tftp Server
     Loaded: loaded (/usr/lib/systemd/system/tftp.service; indirect; vendor preset: disabled)
     Active: inactive (dead) since Thu 2022-02-10 17:11:30 CST; 1h 56min ago
TriggeredBy: ● tftp.socket
       Docs: man:in.tftpd
    Process: 1586282 ExecStart=/usr/sbin/in.tftpd -s /var/lib/tftpboot (code=exited, status=0/SUCCESS)
   Main PID: 1586282 (code=exited, status=0/SUCCESS)
        CPU: 1.781s

Feb 10 16:47:47 cobb3 systemd[1]: Started Tftp Server.
Feb 10 17:11:30 cobb3 systemd[1]: tftp.service: Deactivated successfully.
Feb 10 17:11:30 cobb3 systemd[1]: tftp.service: Consumed 1.781s CPU time.
```

**Comment by monbasa** [comment]
```
# systemctl status tftp.socket
● tftp.socket - Tftp Server Activation Socket
     Loaded: loaded (/usr/lib/systemd/system/tftp.socket; enabled; vendor preset: disabled)
     Active: active (running) since Thu 2022-02-10 11:28:40 CST; 7h ago
   Triggers: ● tftp.service
     Listen: [::]:69 (Datagram)
      Tasks: 0 (limit: 43337)
     Memory: 8.0K
        CPU: 1ms
     CGroup: /system.slice/tftp.socket

Feb 10 11:28:40 cobb3 systemd[1]: Listening on Tftp Server Activation Socket.
```

**Comment by SchoolGuy** [comment, marked as answer]
Yeah and which file is now not found? It is nice to have all that information but it doesn't help to understand the problem sadly.

**Reply by monbasa** [reply]
1. By default configuration
  DHCP templete

 # Legacy
            if option system-arch = 00:00 {
                filename "pxelinux.0";   >>> original is grub/grub.0
 
                ....

               else
          {
              filename "pxelinux.0";   >>> original is grub/grub.0

Pxe cannot boot from read pxelinux.d/default, and cobbler grub menu item is here. so we can not distro item in menu.

**Reply by monbasa** [reply]
And I change to pxelinux.0, and can be boot from pxelinux.d/default.

**Reply by SchoolGuy** [reply]
Okay but why can't it boot from that configuration? And why can't you just stick to GRUB? It is the recommended way.

**Reply by monbasa** [reply]
I find pxe only boot from grub/grub.0 and legacy.cfg, and the file content only local item.

Do you have a recommended linux system installation software?

**Reply by SchoolGuy** [reply]
I cannot grasp what you are trying to tell me. I am sorry. I would love to help you more but I don't understand it.

**Reply by monbasa** [reply]
Sorry to have troubled you.
I found that cobbler 3.3.1, on pxe boot, is calling /var/lib/tftpboot/grub/grub.0 and the grub information from legacy.cfg.
The only information here is the local entry.

I will try cobbler version 3.3.0.

**Reply by SchoolGuy** [reply]
Wait. If the only entry is the local entry. Did you run a `cobbler sync`? because that means the profile is missing in the boot menu. And in theory we check for such things in our testsuite.

**Reply by monbasa** [reply]
I changed the dhcp configuration as follows：

    # more /etc/dhcp/dhcpd.conf
    #
    # DHCP Server Configuration file.
    #   see /usr/share/doc/dhcp-server/dhcpd.conf.example
    #   see dhcpd.conf(5) man page
    #
    #import netaddr
    
    ddns-update-style interim;
    
    allow booting;
    allow bootp;
    
    ignore client-updates;
    set vendorclass = option vendor-class-identifier;
    
    option system-arch code 93 = unsigned integer 16;
    
    subnet 192.168.80.0 netmask 255.255.255.0 {
         option routers             192.168.80.1;
         option domain-name-servers 192.168.80.1;
         option subnet-mask         255.255.255.0;
         range dynamic-bootp        192.168.80.5 192.168.80.10;
         filename                   "pxelinux.0";
         default-lease-time         21600;
         max-lease-time             43200;
    }

Boot menu will Ok.

**Reply by SchoolGuy** [reply]
Okay do you have now the correct entry with the MAC address under `pxelinux.0`?

**Reply by monbasa** [reply]
Sorry, I've been a little busy with work lately. 
Yes, I can boot through pxe normally now. 
Maybe I don't understand enough about pxe and need to learn more before I can.

**Reply by SchoolGuy** [reply]
Okay so this topic is solved now? If yes please mark the answer you believe helped you solve the problem.
